parse_mem.py

```diff
@@ -1,151 +1,90 @@
-
-        if keynum == 0:
-            # Did not find a valid key type
-            print ("A saved key was not found in %s" % self.inputfile)
-            print ("The user may not have loaded a key or the key loaded is " +
-                   "not supported.")
-            sys.exit(1)
-        else:
-            return
-
-    # Detect type of key and run key creation
-    def create_key(self, output):
-        """Creates key files"""
-
-        output = output + "." + self.type
-
-        if self.type is "rsa":
-            self.create_rsa(output)
-            print ("Creating %s key: %s" % (self.type, output))
-        elif self.type is "dsa":
-            self.create_dsa(output)
-            print ("Creating %s key: %s" % (self.type, output))
-        else:
-            print ("%s key type is not currently supported." % self.type)
-            sys.exit(3)
-
-    def create_dsa(self, output):
-        """Create DSA SSH key file"""
-        if self.mem[0:7] == "ssh-dss":
-            print ("DSA SSH Keys are not currently supported.")
-            self.mem = self.mem[start+size:]
-
-        else:
-            print ("Error: This is not a DSA SSH key file")
-            sys.exit(2)
-
-    def create_rsa(self, output):
-        """Create RSA SSH key file"""
-        if self.mem[0:7] == "ssh-rsa":
-
-            # FIXME: This needs to be cleaned up.
-            start = 10
-            size = self.unpack_bigint(self.mem[start:(start+2)])
-            start += 2
-            n = self.unpack_bigint(self.mem[start:(start+size)])
-            start = start + size + 2
-            size = self.unpack_bigint(self.mem[start:(start+2)])
-            start += 2
-            e = self.unpack_bigint(self.mem[start:(start+size)])
-            start = start + size + 2
-            size = self.unpack_bigint(self.mem[start:(start+2)])
-            start += 2
-            d = self.unpack_bigint(self.mem[start:(start+size)])
-            start = start + size + 2
-            size = self.unpack_bigint(self.mem[start:(start+2)])
-            start += 2
-            c = self.unpack_bigint(self.mem[start:(start+size)])
-            start = start + size + 2
-            size = self.unpack_bigint(self.mem[start:(start+2)])
-            start += 2
-            p = self.unpack_bigint(self.mem[start:(start+size)])
-            start = start + size + 2
-            size = self.unpack_bigint(self.mem[start:(start+2)])
-            start += 2
-            q = self.unpack_bigint(self.mem[start:(start+size)])
-
-            e1 = d % (p - 1)
-            e2 = d % (q - 1)
-
-            self.mem = self.mem[start+size:]
-
-        else:
-            print ("Error: This is not a RSA SSH key file")
-            sys.exit(2)
+                print("{}key type is not currently supported.".format(key_type))
+                return
+
+    def get_rsa_key(self, output_file_path):
+        """
+        unpack all rsa data.
+        """
+        def get_value(x):
```

Comment:
`get_value` is solely used for L79-L84, could these possibly be squashed to something that returns a tuple of the key values from the file contents?